### DISCUSSION TITLE
Decorator to listen for `create` and `update` events on all entities

### DISCUSSION THREAD

**Discussion by hakimio** [dxfrontier/cds-ts-dispatcher#57] [Q&A] [answered]
Is it possible to listen for `afterCreate` and `afterUpdate` events on all entities using `cds-ts-dispatcher` decorators? Similar to `srv.after('CREATE', '*', req => {...})`.

**Comment by dragolea** [comment, marked as answer]
Hi Tomas,
Option1: It should be possible, but probably with the help of a new decorator like EntityHandler which should not point to any entity.

Option 2: Overload on EntityHandler to accept star ( all ) then crate new decorators like AfterCreateAll() AfterUpdateAll() ... 

Do you need only for AfterCreate, AfterUpdate all only ?

If we create for After(Create,Update,Read,Delete) then I will implement also for Before and On.

**Reply by hakimio** [reply]
I might need `AfterDelete` as well.
I think `EntityHandler()` overload might be the better option. Sth like `EntityHandler(CDS_DISPATCHER.ALL_ENTITIES)`.
In the long run `Before` and `On` might be useful. For now, my idea is to listen to `After` events to broadcast the events as websocket messages.

**Reply by dragolea** [reply]
Cool, 
Do not forget to create an issue and just write very small summary of what we've discussed.
